receiver/hostmetricsreceiver/hostmetrics_receiver.go

```diff
@@ -48,42 +54,101 @@ func NewHostMetricsReceiver(
 			return nil, fmt.Errorf("host metrics scraper factory not found for key: %s", key)
 		}
 
-		scraper, err := factory.CreateMetricsScraper(ctx, logger, cfg, consumer)
+		scraper, err := factory.CreateMetricsScraper(ctx, logger, cfg)
 		if err != nil {
 			return nil, fmt.Errorf("cannot create scraper: %s", err.Error())
 		}
 		scrapers = append(scrapers, scraper)
 	}
 
-	hmr := &Receiver{
+	hmr := &receiver{
 		config:   config,
 		scrapers: scrapers,
+		consumer: consumer,
 	}
 
 	return hmr, nil
 }
 
-// Start begins scraping host metrics based on the OS platform.
-func (hmr *Receiver) Start(ctx context.Context, host component.Host) error {
+// Start initializes the underlying scrapers and begins scraping
+// host metrics based on the OS platform.
+func (hmr *receiver) Start(ctx context.Context, host component.Host) error {
+	ctx, hmr.cancel = context.WithCancel(ctx)
+
+	go func() {
+		hmr.initializeScrapers(ctx, host)
+		hmr.startScrapers(ctx)
```

Comment:
You are starting an async op here so you need to use a different context because the given context may get cancelled by the caller for example to implement deadline for start operation.

Reply:
A new context for the async operation is started inside the `startScrapers` function. The context passed via this parameter is just used to determine when the receiver's ticker should be stopped - via `ctx.Done()`